### DISCUSSION TITLE
Coupling Concentration Variable "c" to Charge Conservation

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#25490] [Q&A Modules: Phase field]
Hello,

In my PF simulation, the state variable concentration, c(x,y,z,t) is one parameter necessary for computing electrical conductivity.

Electrical conductivity shows up in charge conservation, where I use the HeatConduction kernel and the diffusion_coefficient is established by either a constant in the kernel or in Material using electrical_conductivity.

Can I define an expression including c directly in the HeatConduction or do I need a custom kernel that provides the coupling? This would be similar to using expression and coupled_variables in Materials but directly in the HeatConduction kernel.

Thank you.

John Sevic

**Comment by lindsayad** [comment]
You could either modify `HeatConduction` or create your own kernel. If you are coupling in a variable you'd want to make sure that gets into the Jacobian and currently `HeatConduction` only implements a Jacobian on-diagonal

**Comment by jsevic** [comment]
OK thanks, so it sounds like I could not, for example, create an auxilary varaible say sigma = f(c) and then couple that into HeatConduction by setting diffusion_coefficent to sigma?

I can create my own kernel, it sounds like it would have a the diffusion operator and then I would couple into that an external variable and this could be c, for example?

Thx!

John Sevic

**Reply by GiudGiud** [reply]
is c really only a functioon of x, y z and t ?
if so then you can use:
- a function (functionDiffusion)
https://mooseframework.inl.gov/source/kernels/FunctionDiffusion.html
- an aux-variable
- a material property (with HeatConduction)

They all work.
In general though, if c depends on a nonlinear variable you cannot use Functions or AuxVariables (unless you are not using AD for computing the Jacobian).
